components/NavDrawer/index.jsx

```diff
@@ -119,82 +122,173 @@ class NavDrawer extends Component {
+          props={{
+            ...props,
+            productCount: cartProductCount,
+          }}
         >
-          <I18n.Text string="navigation.cart" />
-        </CartItem>
+          <CartItem
+            href={CART_PATH}
+            icon={ShoppingCartIcon}
+            count={cartProductCount}
+            close={this.handleClose}
+          >
+            <I18n.Text string="navigation.cart" />
+          </CartItem>
+        </Portal>
+        <Portal name={cartPortals.NAV_MENU_CART_AFTER} />
 
         <Divider close={this.handleClose} />
 
         {showQuickLinks && this.renderEntries(entries.quicklinks)}
         {showQuickLinks && <Divider close={this.handleClose} />}
 
-        <Item href={`${PAGE_PATH}/shipping`} icon={LocalShippingIcon} close={this.handleClose}>
-          <I18n.Text string="navigation.shipping" />
-        </Item>
+        <Portal name={commonPortals.NAV_MENU_SHIPPING_BEFORE} />
+        <Portal name={commonPortals.NAV_MENU_SHIPPING} props={props}>
+          <Item href={`${PAGE_PATH}/shipping`} icon={LocalShippingIcon} close={this.handleClose}>
+            <I18n.Text string="navigation.shipping" />
+          </Item>
+        </Portal>
+        <Portal name={commonPortals.NAV_MENU_SHIPPING_AFTER} />
 
-        <Item href={`${PAGE_PATH}/payment`} icon={CreditCardIcon} close={this.handleClose}>
-          <I18n.Text string="navigation.payment" />
-        </Item>
+        <Portal name={commonPortals.NAV_MENU_PAYMENT_BEFORE} />
+        <Portal name={commonPortals.NAV_MENU_PAYMENT} props={props}>
+          <Item href={`${PAGE_PATH}/payment`} icon={CreditCardIcon} close={this.handleClose}>
+            <I18n.Text string="navigation.payment" />
+          </Item>
+        </Portal>
+        <Portal name={commonPortals.NAV_MENU_PAYMENT_AFTER} />
 
         <Divider close={this.handleClose} />
 
-        <Item href={`${PAGE_PATH}/terms`} icon={DescriptionIcon} close={this.handleClose}>
-          <I18n.Text string="navigation.terms" />
-        </Item>
+        <Portal name={commonPortals.NAV_MENU_TERTMS_BEFORE} />
+        <Portal name={commonPortals.NAV_MENU_TERTMS} props={props}>
+          <Item href={`${PAGE_PATH}/terms`} icon={DescriptionIcon} close={this.handleClose}>
+            <I18n.Text string="navigation.terms" />
+          </Item>
+        </Portal>
+        <Portal name={commonPortals.NAV_MENU_TERTMS_AFTER} />
 
-        <Item href={`${PAGE_PATH}/privacy`} icon={SecurityIcon} close={this.handleClose}>
-          <I18n.Text string="navigation.privacy" />
-        </Item>
+        <Portal name={commonPortals.NAV_MENU_PRIVACY_BEFORE} />
+        <Portal name={commonPortals.NAV_MENU_PRIVACY} props={props}>
+          <Item href={`${PAGE_PATH}/privacy`} icon={SecurityIcon} close={this.handleClose}>
+            <I18n.Text string="navigation.privacy" />
+          </Item>
+        </Portal>
+        <Portal name={commonPortals.NAV_MENU_PRIVACY_AFTER} />
 
         {showReturnPolicy && (
```

Comment:
The before and after portals should not depend on the return policy being shown or not.
There may be a case where an extension needs to come before the return policy. If it's then disabled then this portal is closed.